q2_quality_filter/_transformer.py

```diff
@@ -45,3 +50,13 @@ def _2(ff: QualityFilterStatsFmt) -> pd.DataFrame:
 @plugin.register_transformer
 def _3(ff: QualityFilterStatsFmt) -> qiime2.Metadata:
     return qiime2.Metadata(_stats_to_df(ff))
+
+
+@plugin.register_transformer
+def _4(ff: _ReadDirectionUnion) -> SingleLanePerSampleSingleEndFastqDirFmt:
+    return ff.format
+
+
+@plugin.register_transformer
```

Comment:
this is necessary for the type match--it will transform to single-end if the `format` attribute is single-end and will transform to paired-end if the `format` attribute is paired-end

Reply:
So what are the reverse reads if you transform from single end to paired end?

Reply:
Single end doesn't get transformed to paired end. The output format has to however be a type that is in theory transformable to both single-end and paired-end. It will only get transformed to the correct type (single or paired) based on the type map.